src/api.ts

```diff
@@ -0,0 +1,24 @@
+import { IGroup, ISchedule } from "@/interfaces";
+
+import axios from "axios";
+
+const instance = axios.create({ // здесь мы создаем отдельное его "состояние"
+  baseURL: "https://spbgti-tools-schedule-staging.herokuapp.com/api",
+  timeout: 10000
+});
+
+export async function getGroups(): Promise<IGroup[]> {
+  return (await instance.get("/groups")).data as IGroup[];
+}
+
+export async function getScheduleById(id: string, num: string, year: string, semester: string): Promise<ISchedule[]> {
+  let params = {
+    group: id,
```

Comment:
Our API allows request schedule for a group by `id` or by `number`. I prefer explicit way with group_id instead group_id+group_number